lib/auth/password.go

```diff
@@ -195,6 +196,25 @@ func (a *Server) checkPasswordWOToken(user string, password []byte) error {
 		return trace.BadParameter(errMsg)
 	}
 
+	// At this point, we know that the user provided a correct password, so we may
+	// stop worrying about timing attacks against the user existence or password.
+	// Now mark the password state as SET to gradually transition those users for
+	// whom it's not known.
+	_, err = a.UpdateAndSwapUser(context.TODO(), user, true /* withSecrets */, func(u types.User) (bool, error) {
```

Comment:
Nit: Looking at the call sites of `checkPasswordWOToken` it looks like `context.Context` is available in all of them, we should really add this as a parameter to this function rather than using TODO here - we've got a general background ticket to try and propagate `context.Context` more and introducing it where we can to parameters makes a big difference if everyone does it.